src/main/extension/extension_load.cpp

```diff
@@ -70,15 +84,40 @@ bool ExtensionHelper::TryInitialLoad(DBConfig &config, FileOpener *opener, const
 
 		auto signature_offset = handle->GetFileSize() - signature.size();
 
+		const idx_t maxLenChunks = 1024 * 1024;
+		const idx_t numChunks = (signature_offset - 1) / maxLenChunks;
+		std::vector<std::string> chunks(numChunks);
+		std::vector<idx_t> splits(numChunks + 1);
+
+		splits.back() = signature_offset;
+		for (idx_t i = 0; i < chunks.size(); i++)
+			splits[i] = maxLenChunks * i;
+
+#ifndef DUCKDB_NO_THREADS
+		std::deque<std::thread> threads;
+		for (idx_t i = 0; i < numChunks; i++)
+			threads.push_back(std::thread(computeSHA256, handle.get(), splits[i], splits[i+1], &chunks[i]));
+
+		for (idx_t i = 0; i < numChunks; i++)
+			threads[i].join();
+#else
+		for (idx_t i = 0; i < numChunks; i++)
+			computeSHA256(handle.get(), splits[i], splits[i + 1], &chunks[i]);
+#endif // DUCKDB_NO_THREADS
+
 		string file_content;
-		file_content.resize(signature_offset);
-		handle->Read((void *)file_content.data(), signature_offset, 0);
+		file_content.resize(256 * numChunks);
+		for (idx_t i = 0; i < numChunks; i++) {
+			for (idx_t j = 0; j < 256; j++)
```

Comment:
Can we use a `memcpy` here instead, and move `256` to a named constexpr?

Reply:
Moved to reserve length + string concatenation, that should be the same but seems clearer.